salt/utils/azurearm.py

```diff
@@ -316,9 +316,9 @@ def compare_list_of_dicts(old, new, convert_id_to_name=None):
         for key in val:
             local_val = val[key]
             if key in convert_id_to_name:
-                remote_val = val.get(key, {}).get("id", "").split("/")[-1]
+                remote_val = remote_configs.get(key, {}).get("id", "").split("/")[-1]
```

Comment:
Before this was using `remote_configs[idx].get(key...`. It's actually super confusing because it was doing:

```
for idx in range(len(local_configs)):
    for key in local_configs[idx]:
        ...
             remote_val = remote_configs[idx].get(
```
which appears that these lists were assumed to be of equal length? :thinking: But if they were equal length, `zip` would be more appropriate.

I think dictdiffer might actually be better than this approach :thinking:

Reply:
I don't know if they are equal or not.

Reply:
Using enumerate for now to get closer to the original logic.